CommonApplication.cmake

```diff
@@ -2,87 +2,89 @@
 #               2015 Raphael.Dumusc@epfl.ch
 
 # Configures the build for a simple application:
-#   common_application(<Name>)
+#   common_application(<Name> [GUI] [EXAMPLE])
+#
+# Arguments:
+# * GUI: if set, build cross-platform GUI application
+# * EXAMPLE: install all sources in share/Project/examples/Name
 #
 # Input:
 # * NAME_SOURCES for all compilation units
 # * NAME_HEADERS for all internal header files
+# * NAME_SHADERS for all internal shader files (see StringifyShaders.cmake)
 # * NAME_LINK_LIBRARIES for dependencies of name
 # * NAME_OMIT_CHECK_TARGETS do not create cppcheck targets
 # * ARGN for optional add_executable parameters
 # * Optional Qt support:
 # ** NAME_MOC_HEADERS list of all moc input headers
 # ** NAME_UI_FORMS list of all .ui input files
 # ** NAME_RESOURCES list of all .qrc resource files
-#
-# Cross-platform wrapper around common_application() for GUI applications:
-#   common_gui_application(<Name>)
-#
-# Optional OSX bundle support (in conjunction with MACOSX_BUNDLE argument):
-# * NAME_ICON optional .icns file
-# * NAME_COPYRIGHT optional copyright notice
+# * NAME_DATA files for share/Project/data (in binary and install dir)
+# * NAME_ICON optional .icns file (Mac OS GUI applications only)
+# * NAME_COPYRIGHT optional copyright notice (Mac OS GUI applications only)
 #
 # Builds Name application and installs it.
 
 include(AppleCheckOpenGL)
 include(CommonCheckTargets)
 include(CommonQtSupport)
+include(CMakeParseArguments)
+include(StringifyShaders)
 
-# applying CMAKE_C(XX)_FLAGS to add_executable only works from parent scope, hence
-# the macro calling the function _common_application
+# applying CMAKE_C(XX)_FLAGS to add_executable only works from parent
+# scope, hence the macro calling the function _common_application
 macro(COMMON_APPLICATION Name)
   common_compiler_flags()
   _common_application(${Name} ${ARGN})
 endmacro()
 
 function(_common_application Name)
+  set(OPT_NAMES GUI EXAMPLE)
+  set(ARG_NAMES)
+  set(ARGS_NAMES)
+  cmake_parse_arguments(THIS "${OPT_NAMES}" "${ARG_NAMES}" "${ARGS_NAMES}"
```

Comment:
Style:
* remove _NAMES postfix (too many 'name' variables around already)
* use same variable names as in https://cmake.org/cmake/help/v3.2/module/CMakeParseArguments.html ?
* use '_' prefix for local variables